libcxx/include/__undef_macros

```diff
@@ -26,3 +26,7 @@
 #ifdef erase
 #  undef erase
 #endif
+
+#ifdef __noinline__
```

Comment:
Please add a comment about where the problematic macro comes from (cuda).

Reply:
We do that in `system_reserved_names.gen.py`, this file does not have comments for other macros. Do you really want me to break the pattern? Or should I also add comments for other macros in this file?

Reply:
I don't have a strong preference.

Reply:
OK, then I'll keep the existing pattern.